lib/mongoid/contextual/atomic.rb

```diff
@@ -163,10 +175,62 @@ def unset(*args)
         view.update_many("$unset" => Hash[fields])
       end
 
+      # Performs an atomic $min update operation on the given field or fields.
+      # Each field will be set to the minimum of [current_value, given value].
+      # This has the effect of making sure that each field is no
+      # larger than the given value; in other words, the given value is the
+      # effective *maximum* for that field.
+      #
+      # @note Because of the existence of
+      #   Mongoid::Contextual::Aggregable::Mongo#min, this method cannot be
+      #   named #min, and thus breaks that convention of other similar methods
+      #   of being named for the MongoDB operation they perform.
+      #
+      # @example Set "views" to be no more than 100.
+      #   context.set_min(views: 100)
+      #
+      # @param [ Hash ] fields The fields with the maximum value that each
+      #   may be set to.
+      #
+      # @return [ nil ] Nil.
+      def set_min(fields)
+        view.update_many("$min" => collect_operations(fields))
+      end
+      alias :clamp_upper_bound :set_min
+
+      # Performs an atomic $max update operation on the given field or fields.
+      # Each field will be set to the maximum of [current_value, given value].
+      # This has the effect of making sure that each field is no
+      # smaller than the given value; in other words, the given value is the
+      # effective *minimum* for that field.
+      #
+      # @note Because of the existence of
+      #   Mongoid::Contextual::Aggregable::Mongo#max, this method cannot be
+      #   named #max, and thus breaks that convention of other similar methods
+      #   of being named for the MongoDB operation they perform.
+      #
+      # @example Set "views" to be no less than 100.
+      #   context.set_max(views: 100)
+      #
+      # @param [ Hash ] fields The fields with the minimum value that each
+      #   may be set to.
+      #
+      # @return [ nil ] Nil.
+      def set_max(fields)
+        view.update_many("$max" => collect_operations(fields))
+      end
+      alias :clamp_lower_bound :set_max
```

Comment:
In a separate PR we should add a `#clamp` method which combines $max and $min in to one atomic operation:

https://ruby-doc.org/core-3.1.1/Comparable.html#method-i-clamp

To be congruent with Ruby, args can be Array<min, max> OR a Range in the form `field: [-10, 30]` or `field: -10..30`. Ruby supports one-sided ranges for this as well.